subprojects/core/src/integTest/groovy/org/gradle/initialization/buildsrc/BuildSrcIncludedBuildIntegrationTest.groovy

```diff
@@ -32,4 +35,113 @@ class BuildSrcIncludedBuildIntegrationTest extends AbstractIntegrationSpec {
         then:
         failure.assertHasDescription("Cannot include build 'child' in build 'buildSrc'. This is not supported yet.")
     }
+
+    @ToBeFixedForConfigurationCache(because="composite build")
+    def "buildSrc can apply plugins contributed by other included builds"() {
+        file("buildSrc/build.gradle") << """
+            plugins {
+                id "test-plugin"
+            }
+        """
+
+        writePluginTo(file("included"))
+
+        settingsFile << """
+            includeBuild("included")
+        """
+        when:
+        succeeds("help")
+        then:
+        outputContains("test-plugin applied to :buildSrc")
+    }
+
+    @ToBeFixedForConfigurationCache(because="composite build")
+    def "buildSrc can apply plugins contributed by other included builds from CLI"() {
+        file("buildSrc/build.gradle") << """
+            plugins {
+                id "test-plugin"
+            }
+        """
+
+        writePluginTo(file("included"))
+        when:
+        succeeds("help", "--include-build=included")
+        then:
+        outputContains("test-plugin applied to :buildSrc")
+    }
+
+    @ToBeFixedForConfigurationCache(because="composite build")
+    def "buildSrc can apply settings plugins contributed by other included builds"() {
+        file("buildSrc/settings.gradle") << """
+            plugins {
+                id "test-settings-plugin"
+            }
+        """
+
+        def pluginBuilder = new PluginBuilder(file("included"))
+        pluginBuilder.addSettingsPlugin("println 'test-settings-plugin applied to ' + settings.gradle.publicBuildPath.buildPath")
+        pluginBuilder.prepareToExecute()
+
+        settingsFile << """
+            includeBuild("included")
+        """
+        when:
+        succeeds("help")
+        then:
+        outputContains("test-settings-plugin applied to :buildSrc")
+    }
+
+    @ToBeFixedForConfigurationCache(because="composite build")
+    def "buildSrc can apply plugins contributed by nested included build"() {
+        file("buildSrc/build.gradle") << """
+            plugins {
+                id "test-plugin"
+            }
+        """
+
+        writePluginTo(file("included/nested"))
+
+        settingsFile << """
+            includeBuild("included")
+        """
+        file("included/settings.gradle") << """
+            includeBuild("nested")
+        """
+        when:
+        succeeds("help")
+        then:
+        outputContains("test-plugin applied to :buildSrc")
+    }
+
+    @NotYetImplemented
+    def "plugins from buildSrc can be used by other included builds"() {
```

Comment:
I don't think we ever have to fix this. Unless buildSrc becomes a normal included build at some point in the future.

Users who want something like this right now, should just not use buildSrc, but a normal included build.

We could add a comment about that to the test.